src/types.js

```diff
@@ -57,7 +57,8 @@ export type Location = {
   sourceId: SourceId,
   line: number,
   column: ?number,
-  sourceUrl?: string
+  sourceUrl?: string,
+  source?: Source
```

Comment:
lets try and avoid this. Location is one of our most common types and i dont want to imply that Location could have source